script.embuary.helper/resources/lib/utils.py

```diff
@@ -0,0 +1,736 @@
+'''
+def restartservice(params):
+    execute('NotifyAll(%s, restart)' % ADDON_ID)
+
+
+def calc(params):
+    prop = remove_quotes(params.get('prop', 'CalcResult'))
+    formula = remove_quotes(params.get('do'))
+    result = eval(str(formula))
+    winprop(prop, str(result))
+
+
+def settimer(params):
+    actions = remove_quotes(params.get('do'))
+    time = params.get('time', '50')
+    delay = params.get('delay')
+    busydialog = get_bool(params.get('busydialog', 'true'))
+
+    if busydialog:
+        execute('ActivateWindow(busydialognocancel)')
+
+    xbmc.sleep(int(time))
+    execute('Dialog.Close(all,true)')
+
+    while condition('Window.IsVisible(busydialognocancel)'):
+        xbmc.sleep(10)
+
+    for action in actions.split('||'):
+        execute(action)
+        if delay:
+            xbmc.sleep(int(delay))
+
+
+def encode(params):
+    string = remove_quotes(params.get('string'))
+    prop = params.get('prop', 'EncodedString')
+    winprop(prop, url_quote(string))
+
+
+def decode(params):
+    string = remove_quotes(params.get('string'))
+    prop = params.get('prop', 'DecodedString')
+    winprop(prop, url_unquote(string))
+
+
+def getaddonsetting(params):
+    addon_id = params.get('addon')
+    addon_setting = params.get('setting')
+    prop = addon_id + '-' + addon_setting
+
+    try:
+        setting = xbmcaddon.Addon(addon_id).getSetting(addon_setting)
+        winprop(prop,str(setting))
+    except Exception:
+        winprop(prop, clear=True)
+
+
+def togglekodisetting(params):
+    settingname = params.get('setting', '')
+    value = False if condition('system.getbool(%s)' % settingname) else True
+
+    json_call('Settings.SetSettingValue',
+              params={'setting': '%s' % settingname, 'value': value}
+              )
+
+
+def getkodisetting(params):
+    setting = params.get('setting')
+    strip = params.get('strip')
+
+    json_query = json_call('Settings.GetSettingValue',
+                params={'setting': '%s' % setting}
+                )
+
+    try:
+        result = json_query['result']
+        result = result.get('value')
+
+        if strip == 'timeformat':
+            strip = ['(12h)', ('(24h)')]
+            for value in strip:
+                if value in result:
+                    result = result[:-6]
+
+        result = str(result)
+        if result.startswith('[') and result.endswith(']'):
+           result = result[1:-1]
+
+        winprop(setting,result)
+
+    except Exception:
+        winprop(setting, clear=True)
+
+
+def setkodisetting(params):
+    settingname = params.get('setting', '')
+    value = params.get('value', '')
+
+    try:
+        value = int(value)
+    except Exception:
+        if value.lower() == 'true':
+            value = True
+        elif value.lower() == 'false':
+            value = False
+
+    json_call('Settings.SetSettingValue',
+              params={'setting': '%s' % settingname, 'value': value}
+              )
+
+
+def toggleaddons(params):
```

Comment:
I probably asked before but what is this for? 
Not sure if isn't better to create some sort of whitelist to only allow the execution of this method provided a list of allowed addons. Still, we have this exposed in json anyway...

Reply:
Is allowed. Asked in the past.